### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by krastykovyaz**
#MADE

В работе реализован однослойный линейный классификатор - перцептрон. Для начала необходимо установить библиотеки из папки requirements.txt
Реализованы следующие флаги при запуске:
Это для удобства запуска с пояснительной информацией по флагу --help и в случае некорректного ввода.

usage: ft_perceptron.py [-h] [-t] [-p] [--plot] [--valid-size VALID_SIZE] [-lr LEARNING_RATE] [-e EPOCHS]
                        [--batch-size BATCH_SIZE] [--train-path TRAIN_PATH] [--test-path TEST_PATH]
optional arguments:
  -h, --help            show this help message and exit
  -t, --train           train mode
  -p, --predict         predict mode
  --plot                visualization
  --valid-size VALID_SIZE
                        split validation size (0-1)
  -lr LEARNING_RATE, --learning-rate LEARNING_RATE
  -e EPOCHS, --epochs EPOCHS
                        number epochs
  --batch-size BATCH_SIZE
  --train-path TRAIN_PATH, --train-filename TRAIN_PATH
                        train dataset filename
  --test-path TEST_PATH, --test-filename TEST_PATH
                        test dataset filename

Для гибкости работы классификатора реализованы следующие опции.
Проект полностью логируется в режимах обучения и тестирования с промежуточными и итоговыми метриками.
Также реализован режим отображения изменения loss во время обучения. 
Во время обучения можно остановить процесс, что позволит сохранить модель с текущими весами. То есть это остановит процесс обучения, но при этом не сломает саму модель. Это помогает остановить обучение относительно графиков изменения loss, чтобы модель не пере обучилась. Более селективные настройки позволяют конфигурировать с помощью аргументов количество эпох для обучения, также задать learning_rate. Контроль бачей позволяет не перегружать машину  batch_size, а также размер датасета для вадидации valid_size. Флаг --plot  позволяет визуально оценить процесс обучения. Динамически отображается изменение loss. Веса модели и сама модель записывается в json файл.

### CODE REVIEW

**Review by krastykovyaz — COMMENTED** [review]
everything ok
